Remove base rest handler unconsumed parameter filter to force response parameter validation.

howardhuanghua opened the pull request:
This PR is a follow-up enhancement for https://github.com/elastic/elasticsearch/pull/94136.
Remove base rest handler unconsumed parameter filter and force validating response parameters in rest prepare request layer.

howardhuanghua committed:
Add response parameter filter in rest layer.

howardhuanghua committed:
Merge remote-tracking branch 'origin' into rm_resp_filter

elasticsearchmachine commented:
@howardhuanghua please enable the option "Allow edits and access to secrets by maintainers" on your PR. For more information, [see the documentation](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).

elasticsearchmachine commented:
Pinging @elastic/es-core-infra (Team:Core/Infra)

DaveCTurner reviewed (changes requested):
So yes I think we should validate "response" params up front, there is no point in processing a request and then blowing up when serialising the response because one of these parameters was invalid. But we have to actually do some meaningful validation, we can't just blindly consume the parameters like this PR proposes. For instance we should check that boolean parameters will parse as booleans, `ClusterState$Metric` won't throw an `IllegalArgumentException`, and so on...

howardhuanghua commented:
> But we have to actually do some meaningful validation

OK. I would do case by case validation. That should be a huge project.

DaveCTurner commented:
I think it's not all that big a task, there's only these things that I can see:

- [ ] `boolean` parameters. That's almost all of them, and these are easy to validate
- [ ] `AbstractCatAction#RESPONSE_PARAMS` (some of which are boolean)
- [ ] `?settings_filter` (just a raw string, no real validation needed)
- [ ] `?delimiter` as used in the SQL code

howardhuanghua commented:
OK. I am going to add specific validations.

howardhuanghua committed:
Merge branch 'master' into rm_resp_filter

howardhuanghua committed:
pre-validate response param.

howardhuanghua committed:
Merge remote-tracking branch 'origin' into rm_resp_filter

howardhuanghua committed:
Fix spot issue.

howardhuanghua commented:
Hi @DaveCTurner , sorry for the delay. I have updated this PR and added related pre-validation logic for response paramaters. Would you please help to check again? Thanks.
